scripts/compare_warehouse_tables.py

```diff
@@ -0,0 +1,244 @@
+            DevData AS (
+                SELECT
+                    FARM_FINGERPRINT(FORMAT("%T", allDevData)) AS allRows
+                FROM
+                    `broad-dsde-prod-analytics-dev.warehouse_dev.{table_name}` AS allDevData
+                {timestamp_DevData}
+            ),
+            ProdData AS (
+                SELECT
+                    FARM_FINGERPRINT(FORMAT("%T", allProdData)) AS allRows
+                FROM
+                    `broad-dsde-prod-analytics-dev.warehouse.{table_name}` AS allProdData
+                {timestamp_ProdData}
+            )
+            SELECT
+                Sum(IF(ProdData.allRows IS NULL,1,0)) AS in_Dev_not_Prod,
+                Sum(IF(DevData.allRows IS NULL,1,0)) AS in_Prod_not_Dev
+            FROM
+                DevData
+            FULL OUTER JOIN ProdData
+                ON DevData.allRows = ProdData.allRows
+            WHERE
+                ProdData.allRows IS NULL OR DevData.allRows IS NULL"""
+
+            # Calling big query to get In_Dev_Only and In_Prod_Only extra rows count
+            extra_rows = call_bigquery(bq, extra_rows_query, project)
+
+            if time_count == 4:
+                if extra_rows["in_Prod_not_Dev"].values[0] is not None and extra_rows["in_Dev_not_Prod"].values[0] is not None:
+                    not_matched = True
+                    not_matched_info += " Six Month :"
+                time_iteration = False
+                timestamp_DevData = ""
+                timestamp_ProdData = ""
+                time_count = -1
+
+        # Assigning row relust for tablw without a timestamp
+        if not has_timestamp:
+            row_result = extra_rows
+
+        # Getting Dev and Prod table colunms
+        dev_table = bq.get_table(f"warehouse_dev.{table_name}")
+        prod_table = bq.get_table(f"warehouse.{table_name}")
+        prod_table_columns = ["{0}".format(prod_schema.name) for prod_schema in prod_table.schema]
+        dev_table_columns = ["{0}".format(dev_schema.name) for dev_schema in dev_table.schema]
+
+        # Getting In_Dev_Only and In_Prod_Only count from query
+        dev_table_rows = int(row_result["in_Dev_not_Prod"])
+        prod_table_rows = int(row_result["in_Prod_not_Dev"])
+
+        # Adding Columns_Match? boolean and percentage (if not 100%) to the newrow row
+        if prod_table_columns == dev_table_columns:
+            newrows.append("True")
+        else:
+            newrows.append("False")
+            more_info += f"{table_name} table columns Not Matched on " + (", ".join(list(list(set(prod_table_columns) - set(dev_table_columns)) + list(set(dev_table_columns) - set(prod_table_columns))))) + " \n"
+
+        # Adding Rows_Match boolean and percentage (if not 100%) to the newrow row
+        if dev_table_rows is None:
+            newrows.append("True")
+        else:
+            match_percent = int(100 - (((dev_table_rows + prod_table_rows) / dev_table.num_rows) * 100))
```

Comment:
is this being calculated for each time point? your output only shows one value here, which timepoint is that for?